From 63f53d4031e3ffba4e166ecf34df7e9d11d2b55e Mon Sep 17 00:00:00 2001
From: Philip Withnall <philip.withnall@collabora.co.uk>
Date: Fri, 24 Apr 2015 10:27:41 +0100
Subject: [PATCH 5/5] agent: Fix a utility function being defined unnecessarily

sockaddr_to_string() is not needed on UNIX when compiling without
HAVE_GETIFADDRS. This fixes a compiler warning in that case.
---
 agent/interfaces.c | 56 +++++++++++++++++++++++++++---------------------------
 1 file changed, 28 insertions(+), 28 deletions(-)

diff --git a/agent/interfaces.c b/agent/interfaces.c
index 9a0e69c..0fa2fd7 100644
--- a/agent/interfaces.c
+++ b/agent/interfaces.c
@@ -71,6 +71,34 @@
 #include <net/if.h>
 #include <arpa/inet.h>
 
+#endif /* G_OS_UNIX */
+
+#if (defined(G_OS_UNIX) && defined(HAVE_GETIFADDRS)) || defined(G_OS_WIN32)
+/* Works on both UNIX and Windows. Magic! */
+static gchar *
+sockaddr_to_string (const struct sockaddr *addr)
+{
+  char addr_as_string[INET6_ADDRSTRLEN+1];
+  size_t addr_len;
+
+  switch (addr->sa_family) {
+    case AF_INET: addr_len = sizeof (struct sockaddr_in); break;
+    case AF_INET6: addr_len = sizeof (struct sockaddr_in6); break;
+    default: return NULL;
+  }
+
+  if (getnameinfo (addr, addr_len,
+          addr_as_string, sizeof (addr_as_string), NULL, 0,
+          NI_NUMERICHOST) != 0) {
+    return NULL;
+  }
+
+  return g_strdup (addr_as_string);
+}
+#endif
+
+#ifdef G_OS_UNIX
+
 #ifdef HAVE_GETIFADDRS
 
 GList *
@@ -209,9 +237,6 @@ add_ip_to_list (GList *list, gchar *ip, gboolean append)
 
 #ifdef HAVE_GETIFADDRS
 
-static gchar *
-sockaddr_to_string (const struct sockaddr *addr);
-
 GList *
 nice_interfaces_get_local_ips (gboolean include_loopback)
 {
@@ -461,9 +486,6 @@ GList * nice_interfaces_get_local_interfaces (void)
   return ret;
 }
 
-static gchar *
-sockaddr_to_string (const struct sockaddr *addr);
-
 GList * nice_interfaces_get_local_ips (gboolean include_loopback)
 {
   IP_ADAPTER_ADDRESSES *addresses = NULL, *a;
@@ -638,25 +660,3 @@ gchar * nice_interfaces_get_ip_for_interface (gchar *interface_name)
 #error Can not use this method for retreiving ip list from OS other than unix or windows
 #endif /* G_OS_WIN32 */
 #endif /* G_OS_UNIX */
-
-/* Works on both UNIX and Windows. Magic! */
-static gchar *
-sockaddr_to_string (const struct sockaddr *addr)
-{
-  char addr_as_string[INET6_ADDRSTRLEN+1];
-  size_t addr_len;
-
-  switch (addr->sa_family) {
-    case AF_INET: addr_len = sizeof (struct sockaddr_in); break;
-    case AF_INET6: addr_len = sizeof (struct sockaddr_in6); break;
-    default: return NULL;
-  }
-
-  if (getnameinfo (addr, addr_len,
-          addr_as_string, sizeof (addr_as_string), NULL, 0,
-          NI_NUMERICHOST) != 0) {
-    return NULL;
-  }
-
-  return g_strdup (addr_as_string);
-}
-- 
2.3.2 (Apple Git-55)

